docs/content/docs/dev/table/materialized-table/statements.md

```diff
@@ -326,6 +328,69 @@ ALTER MATERIALIZED TABLE my_materialized_table REFRESH PARTITION (ds='2024-06-28
 <span class="label label-danger">Note</span>
 - The REFRESH operation will start a Flink batch job to refresh the materialized table data.
 
+## AS <select_statement>
+
+```sql
+ALTER MATERIALIZED TABLE [catalog_name.][db_name.]table_name AS <select_statement>
+```
+
+The `AS <select_statement>` clause allows you to modify the query definition for refreshing materialized table. It will first evolve the table's schema using the schema derived from the new query and then use the new query to refresh the table data. It is important to emphasize that, by default, this does not impact historical data.
+
+The modification process depends on the refresh mode of the materialized table:
+
+**Full mode:**
+
+1. Update the `schema` and `query definition` of the materialized table.
+2. The table is refreshed using the new query definition when the next refresh job is triggered:
+- If it is a partitioned table and [partition.fields.#.date-formatter]({{< ref "docs/dev/table/config" >}}#partition-fields-date-formatter) is correctly set, only the latest partition will be refreshed.
+- Otherwise, the table will be overwritten entirely.
+
+**Continuous mode:**
+
+1. Pause the current running refresh job.
+2. Update the `schema` and `query definition` of the materialized table.
+3. Start a new refresh job to refresh the materialized table:
+- The new refresh job starts from the beginning and does not restore from the previous state.
+- The starting offset of the data source is determined by the connector’s default implementation or the `option hint` specified in the query.
```

Comment:
`option hint` -> `dynamic options`, I think we can give a link that can help the user to understand it: https://nightlies.apache.org/flink/flink-docs-master/docs/dev/table/sql/queries/hints/#dynamic-table-options